src/appliance-onboarding-script/appliance_setup/run.py

```
@@ -182,5 +196,27 @@ def _populate_default_values_of_optional_fields_in_config(config):
         arc_vmware_res = ArcVMwareResources(config)
         appliance_setup = ApplianceSetup(config, arc_vmware_res, isAutomated)
         appliance_setup.delete()
+    elif operation == 'collect-logs': 
+        if storageAccountName == None:
+            raise InvalidInputError("storageAccount needs to be provided to collect and upload logs.")
+        log_timestamp = datetime.timestamp(datetime.now())
+        logs_folder = 'storagelogs{}'.format(log_timestamp)
+        
+        try:
+            collectLogs = CollectLogs(logs_folder, config)
+            collectLogs.fetch_onboardinglogs()
+            if getArcApplianceLogs:
+                collectLogs.fetch_arc_appliance_logs()   
+
+        except Exception as e:
+            logging.error('Failed to capture complete logs. Inner exception is {}'.format(e))
+
+        try:
+            uploadLogs = UploadLogs(storageAccountName)
+            container_name = 'scriptlogs'
+            uploadLogs.upload_folder_to_storage(logs_folder, container_name)
+        except Exception as e:
```

Do we not want to raise any exception in case the logs weren't uploaded successfully?

---

Since this is not fatal to onboard of arc, I haven't raised exception and exiting the code. logging it as error for now. LMK if you have any thoughts.